src/odb/src/db/dbModule.cpp

```diff
@@ -543,6 +543,372 @@ dbBlock* dbModule::getOwner()
+  debugPrint(logger,
+             utl::ODB,
+             "replace_design",
+             1,
+             "copyModuleModNets: modBT_map has {} ports, it_map has {} iterms",
+             mod_bt_map.size(),
+             it_map.size());
+  // Make boundary port connections.
+  dbSet<dbModNet> old_nets = old_module->getModNets();
+  dbSet<dbModNet>::iterator net_iter;
+  for (net_iter = old_nets.begin(); net_iter != old_nets.end(); ++net_iter) {
+    dbModNet* old_net = *net_iter;
+    dbModNet* new_net = dbModNet::create(new_module, old_net->getName());
+    if (new_net) {
+      debugPrint(logger,
+                 utl::ODB,
+                 "replace_design",
+                 1,
+                 "Created module mod net {}",
+                 new_net->getName());
+    } else {
+      logger->error(utl::ODB,
+                    460,
+                    "Module mod net {} cannot be created",
+                    old_net->getName());
+    }
+
+    // Connect dbModBTerms to new mod net
+    dbSet<dbModBTerm> mbterms = old_net->getModBTerms();
+    dbSet<dbModBTerm>::iterator mb_iter;
+    for (mb_iter = mbterms.begin(); mb_iter != mbterms.end(); ++mb_iter) {
+      dbModBTerm* old_mbterm = *mb_iter;
+      dbModBTerm* new_mbterm = mod_bt_map[old_mbterm];
+      if (new_mbterm) {
+        new_mbterm->connect(new_net);
+        debugPrint(logger,
+                   utl::ODB,
+                   "replace_design",
+                   1,
+                   "  connected port {} to mod net {}",
+                   new_mbterm->getName(),
+                   new_net->getName());
+      } else {
+        logger->error(utl::ODB,
+                      461,
+                      "Port {} cannot be connected to mod net {} because it "
+                      "does not exist",
+                      old_mbterm->getName(),
+                      new_net->getName());
+      }
+    }
+
+    // Connect iterms to new mod net
+    dbSet<dbITerm> iterms = old_net->getITerms();
+    dbSet<dbITerm>::iterator it_iter;
+    for (it_iter = iterms.begin(); it_iter != iterms.end(); ++it_iter) {
+      dbITerm* old_iterm = *it_iter;
+      dbITerm* new_iterm = it_map[old_iterm];
+      if (new_iterm) {
+        new_iterm->connect(new_net);
+        debugPrint(logger,
+                   utl::ODB,
+                   "replace_design",
+                   1,
+                   "  connected iterm {} to mod net {}",
+                   new_iterm->getName(),
+                   new_net->getName());
+      } else {
+        logger->error(utl::ODB,
+                      462,
+                      "Instance terminal {} cannot be connected to mod net {} "
+                      "because it does not exist",
+                      old_iterm->getName(),
+                      new_net->getName());
+      }
+    }
+  }
+}
+
+void dbModule::copyModuleBoundaryIO(dbModule* old_module,
+                                    dbModule* new_module,
```

Comment:
This is the "guts" of the swapping of the modules. Note that by switching the reference to the moditerm on the new master to be the old moditerm, you will get the "right" external modnet connections on the instance (so you don't have to worry about the modnet connection). Looks good.